File: lc_soln/0875_koko_eating_bananas/solution.cs
public class Solution
{
    public int MinEatingSpeed(int[] piles, int h)
    {
        int l = 1;
        int r = piles.Max();
        int result = r;
        while (l <= r)
        {
            int mid = l + (r - l) / 2;
            int count = 0;
            foreach (int pile in piles)
            {
                count += (pile + mid - 1) / mid;
            }
            if (count <= h)
            {
                result = mid;
                r = mid - 1;
            }
            else
            {
                l = mid + 1;
            }
        }
        return result;
    }
}

File: lc_soln/0875_koko_eating_bananas/solution.rs
struct Solution;

impl Solution {
    pub fn min_eating_speed(piles: Vec<i32>, h: i32) -> i32 {
        let mut l = 1i32;
        let mut r = piles.iter().max().copied().unwrap();
        let mut result = r;
        while l <= r {
            let mid = l + (r - l) / 2;
            let count: i32 = piles.iter().map(|&pile| (pile + mid - 1) / mid).sum();
            if count <= h {
                result = mid;
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        result
    }
}

fn main() {}
